packages/mobile/e2e/src/usecases/NewAccountPhoneVerification.js

```diff
@@ -0,0 +1,252 @@
+        await waitFor(element(by.id('VerificationCode0')))
+          .toExist()
+          .withTimeout(45 * 1000)
+
+        // Check that we've received 3 codes
+        console.log(codes)
+        jestExpect(codes).toHaveLength(3)
+
+        // Enter 3 codes
+        for (let i = 0; i < 3; i++) {
+          await sleep(1000)
+          await waitFor(element(by.id(`VerificationCode${i}`)))
+            .toBeVisible()
+            .withTimeout(15 * 1000)
+          await element(by.id(`VerificationCode${i}`)).typeText(codes[i])
+        }
+
+        // Assert we've arrived at the home screen
+        await waitFor(element(by.id('SendOrRequestBar')))
+          .toBeVisible()
+          .withTimeout(45 * 1000)
+
+        // Assert that phone verification CTA is NOT served
+        try {
+          await element(by.id('CTA/ScrollContainer')).scroll(500, 'right')
+        } catch {}
+        await expect(element(by.text('Confirm Now'))).not.toExist()
+
+        // Assert that correct phone number is present in sidebar
+        await element(by.id('Hamburger')).tap()
+        await expect(element(by.text(`${examplePhoneNumber}`))).toBeVisible()
+      })
+      // Note: (Tom) Skip this test until we have a nightly suite vs pull request suite as it takes a long time
+      // jest.retryTimes(2)
+      it.skip('Then should be able to resend last 2 messages', async () => {
+        // Get Date at start
+        let date = new Date()
+        // Start verification
+        await element(by.text('Start')).tap()
+
+        // Request codes, but wait for all 3 to verify resend codes work
+        const codes = await receiveSms(date)
+        await waitFor(element(by.id('VerificationCode0')))
+          .toExist()
+          .withTimeout(45 * 1000)
+
+        // Assert that we've received 3 codes
+        jestExpect(codes).toHaveLength(3)
+
+        // Input first code
+        await element(by.id(`VerificationCode0`)).replaceText(codes[0])
+
+        // Wait one minute before resending
+        await sleep(60 * 1000)
+        await element(by.text('Resend 2 messages')).tap()
+
+        // Enter pin to start resend
+        date = new Date()
+        await enterPinUi()
+        let secondCodeSet = await receiveSms(date, 2, codes)
+
+        // Assert that we've received at least 2 codes
+        jestExpect(secondCodeSet.length).toBeGreaterThanOrEqual(2)
+
+        // Input codes two codes
+        for (let i = 0; i < 2; i++) {
+          await waitFor(element(by.id(`VerificationCode${i + 1}`)))
+            .toBeVisible()
+            .withTimeout(10 * 1000)
+          await element(by.id(`VerificationCode${i + 1}`)).replaceText(secondCodeSet[i])
+        }
+
+        // Assert we've arrived at the home screen
+        await waitFor(element(by.id('SendOrRequestBar')))
+          .toBeVisible()
+          .withTimeout(30 * 1000)
+
+        // Assert that phone verification CTA is NOT served
+        try {
```

Comment:
Same as above, maybe we can move these checks to a function that's called from both places?